### PR TITLE
chore: track failed deeplinks + nav failures

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [PHIRE-621] 

### Description

Adds some sentry tracking to failed navigation events in hopes of identifying more dead ends and deep link breaks quickly. 

3 types of nav errors now tracked:
- 404s in webviews [here](https://artsynet.sentry.io/issues/5709471899/?project=5867225)
- Deeplinks that go more than 10 seconds without showing a screen [here](https://artsynet.sentry.io/issues/5709735888/?project=5867225)
- Custom not found views - e.g. navigating to an artist that doesn't exist [here](https://artsynet.sentry.io/issues/5705613805/?project=5867225)

### Follow-ups

- [ ] Noticed a bug where sale pages don't error out and just show skeleton view forever if not found: https://artsyproduct.atlassian.net/browse/PHIRE-1115
- [ ] Artworks I believe have custom not found behavior might want to start tracking those as well

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- track navigation failure events - brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-621]: https://artsyproduct.atlassian.net/browse/PHIRE-621?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ca6793fc86747dc5f77ea34cd0cd49f09d485aa3
</p>

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (track navigation failure events - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 69aea9b119f159bb40e6922a87cc2fef1c9b9898
</p>

File: src/app/system/navigation/ModalStack.tsx [hunk 1]
```diff
@@ -61,6 +62,8 @@ export const ModalStack: React.FC = ({ children }) => {
         const currentRoute = __unsafe_mainModalStackRef.current?.getCurrentRoute()
 
         if (currentRoute) {
+          LegacyNativeModules.ARTDeeplinkTimeoutModule.invalidateDeeplinkTimeout()
+
```
Comment (gkartalis) [inline src/app/system/navigation/ModalStack.tsx:66]: should we add a check for this to run only on iOS? What will happen if this code runs on android? Will it crash?

File: src/app/Components/NotFoundFailureView.tsx [hunk 2]
```diff
@@ -1,23 +1,36 @@
...
   const showErrorMessage = __DEV__ || isDevToggleEnabled
 
+  useEffect(() => {
+    if (route) {
+      Sentry.withScope((scope) => {
+        scope.setExtra("route", route)
+        Sentry.captureMessage("Navigation: Not found", "error")
+      })
+    }
+  }, [])
```
Comment (gkartalis) [inline src/app/Components/NotFoundFailureView.tsx:32]: ⭐ 🌟 ⭐ 🌟 ⭐ 🌟

File: src/app/Components/ArtsyWebView.tsx [hunk 3]
```diff
@@ -278,6 +279,16 @@ export const ArtsyWebView = forwardRef<
...
+          onHttpError={(error) => {
+            const nativeEvent = error.nativeEvent
+            if (nativeEvent.statusCode === 404) {
+              Sentry.withScope((scope) => {
+                scope.setExtra("route", nativeEvent.url)
+                scope.setExtra("description", nativeEvent.description)
+                Sentry.captureMessage("Navigation: WebView failed to load URL", "error")
+              })
+            }
+          }}
```
Comment (gkartalis) [inline src/app/Components/ArtsyWebView.tsx:291]: 👏 👏 👏 👏

**Review by gkartalis — APPROVED** [review]
Awesome work, added only a question around the LegacyNativeModule running on Android but other than that ⭐

**Comment by gkartalis** [comment]
Thank you for also adding the sentry links - super helpful - Curious if we do need to take any actions for this: 


Screenshot 2024-08-15 at 15 00 10

**Comment by brainbicycle** [comment]
> Thank you for also adding the sentry links - super helpful - Curious if we do need to take any actions for this:
> 
> Screenshot 2024-08-15 at 15 00 10

i think this is just because I was running locally so there was no associated dsym file but will test again once in beta

File: src/app/system/navigation/ModalStack.tsx (hunk 1 above)
Comment (brainbicycle) [reply src/app/system/navigation/ModalStack.tsx:66]: it is a no op in the native modules, see here: https://github.com/artsy/eigen/blob/69aea9b119f159bb40e6922a87cc2fef1c9b9898/src/app/NativeModules/LegacyNativeModules.tsx#L113 but I can add a check to make it more explicit

**Pull request merged**